### ISSUE TITLE
New rule `vue/quotes`

### ISSUE THREAD

**Issue by szulcus** [vuejs/eslint-plugin-vue#2187]
**Tell us about your environment**

- **ESLint version:** ^8.28.0
- **eslint-plugin-vue version:**  9.14.0
- **Node version:** v16.19.0

**The problem you want to solve.**
This warning, but in vue tamplate:
![image](https://github.com/vuejs/eslint-plugin-vue/assets/48328739/5cadbd26-50f8-4458-ba72-bce4f8202403)
![image](https://github.com/vuejs/eslint-plugin-vue/assets/48328739/69c968a0-c948-41d3-8ba2-94235048916a)

**Your take on the correct solution to problem.**
Similar like: https://eslint.org/docs/latest/rules/quotes

**Comment by FloEdelmann** [comment]
Sounds good, thanks for the suggestion! This should be relatively easy, as we have a reusable function for wrapping core ESLint rules.

Would you like to give it a try? See https://eslint.vuejs.org/developer-guide/#proposing-a-new-rule-or-a-rule-change.
You can take e.g. the [`vue/camelcase` rule](https://eslint.vuejs.org/rules/camelcase.html) and its implementation as a base.

**Comment by szulcus** [comment]
@FloEdelmann I didn't know there was a way to create custom rules. I'd like to try it, but it might take me a while to get to know it. I won't have time for that anytime soon, but maybe later I can help with the new rules. Thanks for the offer!

**Comment by ota-meshi** [comment]
Note that we may need to exclude warnings within directives.

```vue
<tempalte>
  <img v-bind:title="$t(`foo`)" />
</template>
```

**Comment by Bernankez** [comment]
Hi, I just wrote a [rule](https://github.com/Bernankez/eslint-config/blob/master/packages/eslint-plugin-bernankez/src/rules/mustache-interpolation-quotes.ts) for this, but since it's only for my personal use, the test cases are not comprehensive and only consider quotes in mustache tags. I haven't considered quotes within directives. Maybe you can use it as a reference :)

**Comment by FloEdelmann** [comment]
@ota-meshi Why should directives be excluded from this rule? If one enables this rule to enforce using a specific quote style, that style should also be enforced in directives, shouldn't it?

**Comment by ota-meshi** [comment]
> Why should directives be excluded from this rule? 

This is because using `"` to quote a directive enclosed in `"` will break it.

```vue
<template>
  <img v-bind:title="$t(`foo`)" />
</template>
```

> If one enables this rule to enforce using a specific quote style, that style should also be enforced in directives, shouldn't it?

That's true. But in that case, I think we need to implement it ourselves. In other words, I don't think we can implement it by referring to `vue/camelcase` rule.

**Comment by FloEdelmann** [comment]
> This is because using `"` to quote a directive enclosed in `"` will break it.

Oh true, I missed that obvious fact :sweat_smile: 

Can we add a rule configuration option `checkDirectives` (`false` by default) that would enable checking directives, and which would only make sense if the enforced quote style is not double quotes? Would that still be relatively easy to do while extending the core `quotes` rule?

**Comment by ota-meshi** [comment]
My guess is that it's probably easier to implement own rules than it is to extend the core rule 🤔

**Comment by jacekkarczmarczyk** [comment]
> This is because using `"` to quote a directive enclosed in `"` will break it.

It should not exclude directives, but it should just not autofix it if it's not possible. Rule might work fine if I want to single quotes, it could replace ``v-foo="`bar`"`` to `v-foo="'bar'"`

**Comment by OliverBennettdev** [comment]
Hi! I'd like to pick this up. I saw that #2463 was closed due to inactivity and that a new PR is welcome. I'll use the previous implementation as a reference and make sure to address the outstanding review feedback, including the directive test cases.
